packages/webhook/sdk/src/init.ts

```diff
@@ -0,0 +1,54 @@
+import MessageValidator from 'sns-validator';
+import { Environment } from '@imtbl/config';
+
+const validator = new MessageValidator();
+
+const allowedTopicArnPrefix = {
+  [Environment.PRODUCTION]: 'arn:aws:sns:us-east-2:362750628221:',
+  [Environment.SANDBOX]: 'arn:aws:sns:us-east-2:783421985614:'
+};
+
+export const init = async (
+  body: string | Record<string, unknown>,
+  env: Environment,
+  handlers: {
+    zkevmMintRequestUpdated: (event: any) => Promise<void>;
+    others?: (event: any) => Promise<void>;
+  }
+) => {
+  const msg: any = await new Promise((resolve, reject) => {
+    validator.validate(body, (err, message: any) => {
+      if (err) {
+        return reject(err);
+      }
+
+      if (message?.Type === 'SubscriptionConfirmation') {
+        fetch(message.SubscribeURL).then(() => {
+          resolve(message);
+        }).catch((e) => {
+          reject(e);
+        });
+      }
+      return resolve(message);
+    });
+  });
+
+  // check for topic arn prefix
+  if (!msg.TopicArn.startsWith(allowedTopicArnPrefix[env])) {
```

Comment:
is there a way to check this first before confirming subscription? Otherwise the partner may confirm a subscription from a malicious actor

Reply:
aha... right right.